drift_flutter/lib/src/connect.dart

```diff
@@ -67,10 +68,41 @@ final class DriftNativeOptions {
   /// a custom database path in another directory.
   final Future<String> Function()? databasePath;
 
+  /// The directory where the Drift database files are stored.
+  ///
+  /// If a custom [databasePath] is provided, this option must not be used.
+  ///
+  /// If no custom [databasePath] is provided and this option is not used, the
+  /// default directory is [DriftDatabaseDirectory.applicationDocumentsDirectory].
+  final DriftDatabaseDirectory? directory;
+
   /// Create drift options effective when opening drift databases on native
   /// platforms.
   const DriftNativeOptions({
     this.shareAcrossIsolates = false,
     this.databasePath,
-  });
+    this.directory,
+  }) : assert(databasePath == null || directory == null,
```

Comment:
We should find a way to nudge users into a applying a reasonable default without breaking their apps silently (a compiler error is annoying, but at some point we want to change the default and, since not everyone is reading changelogs, working code shouldn't suddenly use a different directory).

There are different options that come to mind here, e.g.

1. Adding something like a fallback path that would be considered when looking for existing databases and defaults to `applicationDocumentsDirectory`. When we open the database and don't find it in the intended directory, we check if a file exists in the fallback directory and move it if necessary. That makes changing the default less breaking.
2. Making `directory` a required parameter and also requiring the native options on `driftDatabase`. So users have to specify something, and we recommend specifying `applicationSupportDirectory` _unless_ a published version of the app is already using something else.

I don't like option 1 too much because it makes things messy internally and is still breaking (users might now expect the path in other scenarios, e.g. when sharing the database with some native code also opening it).
For the required parameter, we'd probably have to remove the `databasePath` argument from the default constructor and add a named constructor (e.g. `withCustomPath`) that has that argument but no `directory` argument.
What are your thoughts on this?

Reply:
Do you see any terrible downsides of the second approach @dickermoshe? Otherwise I'd start to prepare that in the next days so that we can do a release soon.

Reply:
i would love for there to be a sensible default, but i guess that ship has sailed. we can't change the default and expect people to notice. we can't require people to enter the native options, because they may be making a web only app

Reply:
seems like both approaches have issues